pkg/asset/tls/boundsasigningkey.go

```diff
@@ -0,0 +1,59 @@
+package tls
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// BoundSASigningKey contains a user provided key and public parts for the
+// service account signing key used by kube-apiserver.
+// This asset does not generate any new content and only loads these files from disk
+// when provided by the user.
+type BoundSASigningKey struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*BoundSASigningKey)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*BoundSASigningKey) Name() string {
+	return "User-provided Service Account Signing key"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*BoundSASigningKey) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the CloudProviderConfig.
+func (*BoundSASigningKey) Generate(dependencies asset.Parents) error { return nil }
+
+// Files returns the files generated by the asset.
+func (sk *BoundSASigningKey) Files() []*asset.File {
+	if len(sk.FileList) == 0 {
+		return []*asset.File{}
+	}
+	return sk.FileList
+}
+
+// Load reads the private and public key from the disk.
+// If only one of the private or public key are provided, the loads fails with error.
```

Comment:
This comment does not match the actual behavior of the function. The actual behavior is that, if one of the files does not exist, then the other file is ignored.

Reply:
fixed to match the documentation in eb2f689